### PR TITLE
Navigator:Add animation to resetTo method

### PR DESCRIPTION
**Pull request by CntChen**
The previous `resetTo` method has inconsistent transition animation.
When route stack only contain one route, `resetTo(newRoute)` will immediately render new scene without animation, otherwise has scene transition animation (by using `_popN`);

I think `resetTo` method better have animation to provide consistent user experience. Transition without animation can use `immediatelyResetRouteStack` or `replace`.

Related topic:[`PR#1300`](https://github.com/facebook/react-native/pull/1300)
@ericvicenti @pwmckenna

### CODE REVIEW

**Comment by brentvatne** [comment]
@CntChen - can you rebase this against master?

@ericvicenti - when you get a chance, seems good for review, cc @hedgerwang

**Comment by DannyvanderJagt** [comment]
@CntChen This is what i was looking for :) I have replaced the default code with your code in my RN project and everything seems to work perfect.

File: Libraries/CustomComponents/Navigator/Navigator.js [hunk 1]
```diff
@@ -1028,13 +1028,35 @@ var Navigator = React.createClass({
...
+    var nextAnimationConfigStack = activeAnimationConfigStack.concat([
+      this.props.configureScene(route, nextStack),
+    ]);
+    this._emitWillFocus(nextStack[destIndex]);
+    this.setState({
+      routeStack: nextStack,
+      sceneConfigStack: nextAnimationConfigStack,
+    }, () => {
+      this._enableScene(destIndex);
+      this._transitionTo(destIndex, null, null, () => {
```
Comment (chirag04) [inline Libraries/CustomComponents/Navigator/Navigator.js]: If resetTo will now `push` the route first and then update the stack then y not reuse `push` function that can take a callback which is called after the transition.

Something like:
```
push: function(route, cb) {
    ....
    this._transitionTo(destIndex, null, null, cb);
}
```
```
resetTo: function(route) {
   this.push(route, () => {
       this.setState(.....)
   }):
}
```

Comment (CntChen) [inline Libraries/CustomComponents/Navigator/Navigator.js]: Thanks for your suggest. I had updated codes.

**Comment by ericvicenti** [comment]
@facebook-github-bot shipit

**Comment by mkonicek** [comment]
Sorry about the (long) delay in code review @CntChen! The patch doesn't apply anymore. Can you rebase please?

**Comment by CntChen** [comment]
I had rebase this pull request. @mkonicek 
It's my first pull request in github and I am not familiar with github workflow. If something wrong with this pull request, please let message to me.

**Comment by mkonicek** [comment]
No problem @CntChen, merging again.

**Comment by facebook-github-bot-7** [comment]
@CntChen updated the pull request.

**Comment by ademars94** [comment]
I am trying to incorporate this in a project I'm working on, but when I use resetTo to navigate from my Login page to my Main page, I lose the NavigationBar's Title on my Main page.  It looks like it's there while _push is running, and then when the resetTo setState callback fires, it disappears.

Any insight into what might be going on here?  Any help is appreciated.

**Comment by ademars94** [comment]
@CntChen

**Comment by ademars94** [comment]
Okay, I managed to resolve this problem on my own.  

I had to call `this._navBar.immediatelyRefresh();` in the callback for the `setState` call in `resetTo`, which brought back my title.

**Comment by CntChen** [comment]
good. submit a PR of your solution if you have time. @ademars94

**Pull request closed without merging**